In [1]:
import pandas as pd
url = "https://raw.githubusercontent.com/mricardo89/data-mining/main/Unit-II/datasets/titanic.csv"
df = pd.read_csv(url)
df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [5]:
skew = df['Age'].skew()
print(f'{skew}')
kurt = df['Fare'].kurt()
print(f'{kurt}')
cross = pd.crosstab(df['Pclass'], df['Survived'], margins=True)
print(f'{cross}')


0.38910778230082704
33.39814088089868
Survived    0    1  All
Pclass                 
1          80  136  216
2          97   87  184
3         372  119  491
All       549  342  891


Asimetría (Skewness): Investiguen cómo aplicar la función .skew() en la columna Age. Si el resultado es positivo o negativo, ¿qué significa físicamente sobre la edad de los pasajeros en el barco? (¿Había más jóvenes o más ancianos?).

Viendo que da un resultado de 0.3891, esto significa que hay mas personas jovenes, ya que la cola de distribuicion se estira a la derecha, osea edades mayores, lo que nos dice que la mayoria esta en la izquierda que son edades jovenes

Curtosis (Kurtosis): Investiguen cómo aplicar la función .kurt() en la columna Fare. ¿Qué significa este número sobre el "grosor" de las colas de la distribución? ¿Confirma esto la existencia de los outliers que encontramos hoy en clase?

Se obtiene el valor de 33.398, que es un numero alto positivo, esto significa que la distribucion tiene colas pesadas o gruesas. Confirmando una existencia de outliers, inficando que hay datos extremos o alejados del promedio

Tablas Cruzadas (Cross-Tabulation): Investiguen la función pd.crosstab(). Generen una tabla cruzada entre la clase del pasajero (Pclass) y si sobrevivió (Survived). Esta técnica de sumarización matricial es el equivalente a una Tabla Dinámica en Excel. ¿Qué patrón logran visualizar rápidamente al cruzar ambas variables?

Al momento de cruzar Pclass con Survived, el patron dice que la clase social pudo haber tenido algo que ver con salvarse, ya que los pasajeros de tercera clase fueron los que tuvieron mayores bajas hablando de porcentaje, a comparacion de los de primera clase, donde sobrevivieron la mayoria, en segunda clase fue casi la mitad, mostrando que pasajeros de una clase alta tenia estadisticamente una mayor probabilidad de sobrevivir


